packages/ra-ui-materialui/src/layout/TitleForRecord.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { FC } from 'react';
+import PropTypes from 'prop-types';
+import Title, { TitlePropType, TitleProps } from './Title';
+
+const TitleForRecord: FC<Props> = ({
+    defaultTitle,
+    record,
+    title,
+    className,
+}) =>
+    record ? (
+        <Title
+            title={title}
+            record={record}
+            defaultTitle={defaultTitle}
+            className={className}
+        />
+    ) : null;
+
+interface Props extends TitleProps {
+    record: TitleProps['record'];
```

Comment:
I don't understand this type. It specializes nothing.

Reply:
to make `record` required, but to make things more clear I can add a helper type?

```ts
export type SomeRequired<T, U extends keyof T> = Omit<T, U> &
  Required<Pick<T, U>>;
```

```diff
+ interface Props extends <TitleProps,'record'> {
- interface Props extends TitleProps {
-     record: TitleProps['record'];
```